specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2020-10-20/myworkbooks_API.json

```diff
@@ -0,0 +1,564 @@
+      "required": [
+        "displayName",
+        "category",
+        "serializedData"
+      ],
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "The user-defined name of the private workbook."
+        },
+        "serializedData": {
+          "type": "string",
+          "description": "Configuration of this particular private workbook. Configuration data is a string containing valid JSON"
+        },
+        "version": {
+          "type": "string",
+          "description": "This instance's version of the data model. This can change as new features are added that can be marked private workbook."
+        },
+        "timeModified": {
+          "type": "string",
+          "description": "Date and time in UTC of the last modification that was made to this private workbook definition.",
+          "readOnly": true
+        },
+        "category": {
+          "type": "string",
+          "description": "Workbook category, as defined by the user at creation time."
+        },
+        "tags": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "A list of 0 or more tags that are associated with this private workbook definition"
+        },
+        "userId": {
+          "type": "string",
+          "description": "Unique user id of the specific user that owns this private workbook.",
+          "readOnly": true
+        },
+        "sourceId": {
+          "type": "string",
+          "description": "Optional resourceId for a source resource."
+        },
+        "storageUri": {
+          "type": "string",
+          "description": "BYOS Storage Account URI"
+        }
+      }
+    },
+    "ErrorFieldContract": {
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "Property level error code."
+        },
+        "message": {
+          "type": "string",
+          "description": "Human-readable representation of property-level error."
+        },
+        "target": {
+          "type": "string",
+          "description": "Property name."
+        }
+      },
+      "description": "Error Field contract."
+    },
+    "MyWorkbookError": {
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "Service-defined error code. This code serves as a sub-status for the HTTP error code specified in the response."
+        },
+        "message": {
+          "type": "string",
+          "description": "Human-readable representation of the error."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ErrorFieldContract"
+          },
+          "description": "The list of invalid fields send in request, in case of validation error."
+        }
+      },
+      "description": "Error message body that will indicate why the operation failed."
+    },
+    "ManagedIdentity": {
```

Comment:
>ManagedIdentity [](start = 5, length = 15)

Suggest referring Common Type definitions as follows? 
"$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/Identity"

Reply:
thanks! good call!

Reply:
Remodeled a bit, but in order to use the existing Identity definition, I had to add some enum values.

Reply:
Turns out we cannot have "systemAssigned" as a viable value for the type, so i'll have to use the locally defined identity to avoid a breaking change by removing "SystemAssigned" from the common identity type